### PR TITLE
47 date categories location and interests functionality fix

### LINKED ISSUE #47: Date, Categories, Location and interests functionality fix
**Task title**
Date, Categories, Location and interests functionality fix

**Task description**
- limit the user date input to restricted format dd/mm/yyyy
- provide Location suggestion
- limit the user interests input to a certain range.
- provide restricted range of categories

### PR DESCRIPTION
**Pull request by Khalil-NOUI**
### -provide a drop down of interests, 
        
Screenshot (144)

### -add location suggestions in the location input,

Screenshot (145)

### -support date input with calendar

input calandere

### -filter per date is functional.

P.S: please run npm install before all

#47 

# Related Issue

- Resolve #47

### CODE REVIEW

File: src/components/createEventPageComponents/CreateEvent.jsx [hunk 1]
```diff
@@ -1,24 +1,54 @@
...
+import { interestList } from "../../lib/interestsList";
 
 function CreateEvent() {
+    const [startDate, setStartDate] = useState(new Date());
+    const [selectedInterets, setSelectedInterets] = useState([]);
+    const [loca, setLoca] = useState(null);
+
+    const options = interestList.map((obj) => {
+        return {
+            label: `${obj.title}`,
```
Comment (HachemBouhadede) [inline src/components/createEventPageComponents/CreateEvent.jsx:19]: wasn't an id required ? in the map function ?

File: src/components/createEventPageComponents/CreateEvent.jsx [hunk 2]
```diff
@@ -50,18 +80,16 @@ function CreateEvent() {
...
+                                Interests
                             </label>
-                            <input
-                                type='text'
-                                name='category'
-                                id='category'
-                                className='shadow-sm bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-cyan-600 focus:border-cyan-600 block w-full p-2.5'
-                                placeholder='e.g ..sustaibility'
-                                required
+                            <MultiSelect
```
Comment (HachemBouhadede) [inline src/components/createEventPageComponents/CreateEvent.jsx:88]: can we limite the number of selections with this componenet per example up to 5 max

File: src/components/singlEventPageComponents/EditEvent.jsx [hunk 3]
```diff
@@ -16,39 +33,34 @@ function EditEvent() {
...
     }
 
+    const handleLocationSelect = (selectedLocation) => {
+        setLoca(selectedLocation);
+    };
+
     return (
         <div className='bg-white border rounded-lg shadow relative m-10'>
             <div className='flex items-start justify-between p-5 border-b rounded-t '>
-                <h3 className='text-2xl font-semibold'>Edit Event</h3>
```
Comment (HachemBouhadede) [inline src/components/singlEventPageComponents/EditEvent.jsx:33]: this is the edit event no ?

File: src/components/singlEventPageComponents/EditEvent.jsx [hunk 4]
```diff
@@ -1,13 +1,30 @@
...
 
+import LocationInput from "../reusableComponents/LocationInput";
+import { interestList } from "../../lib/interestsList";
+
 function EditEvent() {
+    const [startDate, setStartDate] = useState(new Date());
+    const [selectedInterets, setSelectedInterets] = useState([]);
+    const [loca, setLoca] = useState(null);
     const router = useRouter();
     const { id } = router.query;
```
Comment (HachemBouhadede) [inline src/components/singlEventPageComponents/EditEvent.jsx:19]: if this is edit the event i think we'll need to state for all inputs to fetch and edit them

**Review by HachemBouhadede — APPROVED** [review]
the edit event page has the same title as the creat event page when you head to it, other then welll done

File: src/components/singlEventPageComponents/EditEvent.jsx (hunk 4 above)
Comment (Khalil-NOUI) [reply src/components/singlEventPageComponents/EditEvent.jsx:19]: Yes, 100%

File: src/components/createEventPageComponents/CreateEvent.jsx (hunk 2 above)
Comment (Khalil-NOUI) [reply src/components/createEventPageComponents/CreateEvent.jsx:88]: yes we can, although it's very difficult to predict the range of interests that an event can fall into.

File: src/components/createEventPageComponents/CreateEvent.jsx (hunk 1 above)
Comment (Khalil-NOUI) [reply src/components/createEventPageComponents/CreateEvent.jsx:19]: i'm not sure i get your question, but in here i'm simply converting the options const which represent the array of interests objects that an event may belong to, into an array of object in which each have the properties of label and value, all to make the multiselect component functional.

File: src/components/singlEventPageComponents/EditEvent.jsx (hunk 3 above)
Comment (Khalil-NOUI) [reply src/components/singlEventPageComponents/EditEvent.jsx:33]: amazing insight i totally missed it, to be fixed asap

File: src/components/createEventPageComponents/CreateEvent.jsx (hunk 2 above)
Comment (HachemBouhadede) [reply src/components/createEventPageComponents/CreateEvent.jsx:88]: i get your point, my intentions is just to limit and the user takes the limit underconsideration that all

**Pull request merged**